docs/extensibility/snippets/MailDevResource/MailKit.Client/MailKitClientServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,133 @@
+    /// connected <see cref="SmtpClient"/> instance for sending emails.
+    /// </summary>
+    /// <param name="builder">
+    /// The <see cref="IHostApplicationBuilder" /> to read config from and add services to.
+    /// </param>
+    /// <param name="connectionName">
+    /// A name used to retrieve the connection string from the ConnectionStrings configuration section.
+    /// </param>
+    /// <param name="configureSettings">
+    /// An optional delegate that can be used for customizing options.
+    /// It's invoked after the settings are read from the configuration.
+    /// </param>
+    public static void AddMailKitClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<MailKitClientSettings>? configureSettings = null) =>
+        AddMailKitClient(
+            builder,
+            MailKitClientSettings.DefaultConfigSectionName,
+            configureSettings,
+            connectionName,
+            serviceKey: null);
+
+    /// <summary>
+    /// Registers 'Scoped' <see cref="MailKitClientFactory" /> for creating
+    /// connected <see cref="SmtpClient"/> instance for sending emails.
+    /// </summary>
+    /// <param name="builder">
+    /// The <see cref="IHostApplicationBuilder" /> to read config from and add services to.
+    /// </param>
+    /// <param name="name">
+    /// The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the
+    /// service and also to retrieve the connection string from the ConnectionStrings configuration section.
+    /// </param>
+    /// <param name="configureSettings">
+    /// An optional method that can be used for customizing options. It's invoked after the settings are
+    /// read from the configuration.
+    /// </param>
+    public static void AddKeyedMailKitClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<MailKitClientSettings>? configureSettings = null)
+    {
+        ArgumentNullException.ThrowIfNull(name);
+
+        AddMailKitClient(
+            builder,
+            $"{MailKitClientSettings.DefaultConfigSectionName}:{name}",
+            configureSettings,
+            connectionName: name,
+            serviceKey: name);
+    }
+
+    private static void AddMailKitClient(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<MailKitClientSettings>? configureSettings,
+        string connectionName,
+        object? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new MailKitClientSettings();
+
+        builder.Configuration
+               .GetSection(configurationSectionName)
+               .Bind(settings);
+
+        settings.ParseConnectionString(
+            builder.Configuration.GetConnectionString(connectionName) ??
+            builder.Configuration[$"{MailKitClientSettings.DefaultConfigSectionName}:Endpoint"]);
```

Comment:
```suggestion
        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
        {
            settings.ParseConnectionString(connectionString);
        }
```

1. You shouldn't throw if there is no conneciton string in the configuration. The `configureSettings?.Invoke(settings);` might go grab it from somewhere else.
2. You don't need to read `builder.Configuration[$"{MailKitClientSettings.DefaultConfigSectionName}:Endpoint"]` here. That already happened on line 83 above.